contracts/campaignAirdrops/ExistingERC20DropCampaign.sol

```diff
@@ -0,0 +1,126 @@
+import "../interfaces/AirdropMerkleProof.sol";
+import "../interfaces/IAirBroFactory.sol";
+
+/// @title Airdrops existing ERC20 tokens for airdrop recipients
+contract ExistingERC20DropCampaign is AirdropMerkleProof {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable rewardToken;
+    address public immutable airbroCampaignFactoryAddress;
+    uint256 public immutable tokenSupply;
+
+    string public airdropType = "ERC20";
+    bool public airdropFunded;
+    bool public merkleRootSet;
+    uint256 internal numberOfClaimers;
+    uint256 public tokensPerClaim;
+    uint256 public airdropExpirationTimestamp;
+    bytes32 public merkleRoot;
+
+    mapping(address => bool) public hasClaimed;
+
+    address internal airdropFunder;
+
+    event MerkleRootChanged(bytes32 merkleRoot, uint256 _numberOfClaimers);
+    event Claimed(address indexed claimer);
+    event AirdropFunded(address contractAddress);
+
+    error AlreadyFunded();
+    error Unauthorized();
+    error AlreadyRedeemed();
+    error AirdropStillActive();
+    error AirdropExpired();
+    error NotEligible();
+    error InsufficientAmount();
+    error InsufficientLiquidity();
+    error MerkleRootAlreadySet();
+
+    modifier onlyAdmin() {
+        if (msg.sender != IAirBroFactory(airbroCampaignFactoryAddress).admin()) revert Unauthorized();
+        _;
+    }
+
+    constructor(
+        address _rewardToken,
+        uint256 _tokenSupply,
+        address _airbroCampaignFactoryAddress
+    ) {
+        rewardToken = IERC20(_rewardToken);
+        tokenSupply = _tokenSupply;
+        airbroCampaignFactoryAddress = _airbroCampaignFactoryAddress;
+    }
+
+    /// @notice Sets the merkleRoot and the number of claimers (also setting the amount each claimer receivers)
+    /// @notice can only be done by admin
+    /// @param _merkleRoot The root hash of the Merle Tree
+    /// @param _numberOfClaimers The number of people eligible to claim
+    function setMerkleRoot(bytes32 _merkleRoot, uint256 _numberOfClaimers) external onlyAdmin {
+        if (merkleRootSet) revert MerkleRootAlreadySet();
+
+        merkleRootSet = true;
+        merkleRoot = _merkleRoot;
+        airdropExpirationTimestamp = block.timestamp + 60 days;
+        numberOfClaimers = _numberOfClaimers;
+        tokensPerClaim = tokenSupply / numberOfClaimers;
+
+        emit MerkleRootChanged(_merkleRoot, _numberOfClaimers);
+    }
+
+    /// @notice Allows the airdrop creator to provide funds for the airdrop reward
+    function fundAirdrop() external {
+        if (airdropFunded) revert AlreadyFunded();
+
+        if (rewardToken.balanceOf(msg.sender) < tokenSupply) revert InsufficientAmount();
+        if (rewardToken.allowance(msg.sender, address(this)) < tokenSupply) revert InsufficientAmount();
+
+        airdropFunded = true;
+        airdropFunder = msg.sender;
+
+        rewardToken.safeTransferFrom(msg.sender, address(this), tokenSupply);
+        emit AirdropFunded(address(this));
```

Comment:
address(this) doesn't make sense to me

Reply:
Requested feature from a frontend dev, still waiting to confirm the reason behind it

Reply:
No update on this from the backend dev who requested it. I suggest we merge it and update when more information is available.